pkg/api/v1/defaults.go

```diff
@@ -228,6 +228,10 @@ func SetDefaults_PersistentVolumeClaim(obj *v1.PersistentVolumeClaim) {
 	if obj.Status.Phase == "" {
 		obj.Status.Phase = v1.ClaimPending
 	}
+	if obj.Spec.VolumeMode == nil {
```

Comment:
Can we default new optional fields?

If you have a 1.7 client talking to a 1.8 server, the 1.8 server is going to default the volumeMode and store it.  Then, when it sends the object back to the 1.7 client, the 1.7 client is going to receive a non-nil value for a field it doesn't understand.

Actually, how would a 1.7 client view objects created with a 1.8 client that uses the new fields?  Or does the client just ignore fields it doesn't know about?

Reply:
@liggitt - can you confirm or elaborate on what @msau42 is saying above?

Reply:
I don't think defaulting is compatible with disabled alpha fields. The server should act as though the field doesn't exist (not persist or return it).

Reply:
ok, I'm going to remove the default for pvc.volumeMode then